Chase_step1b_AssignOutcome_Algorithm4.sas

```diff
@@ -1,162 +1,205 @@
+        	if dx_n>=1 then do i=1 to 10 until (outassgn_pt1 ne 'unk'); *1proc just use that one;
 
-            if hier6(i) in ('100','101','110','111'  '102','112','103','113') then do; 
-                 outassgn_pt1= vname(hier6(i));
-                 *outassgn_pt1_Codetype='DX';
-            end;
+            	if hier6(i) in ('100','101','110','111'  '102','112','103','113') then do; 
+                 	outassgn_pt1= vname(hier6(i));
+            	end;
 
-            else if rx_n>=1 then do; 
-             outassgn_pt1  = 'UAB';
-             *outassgn_pt1 _Codetype = 'RX';  
-            end; 
+            	else if rx_n>=1 then do; 
+             		outassgn_pt1  = 'UAB';
+            	end; 
 
-        end;
-    end; 
+        	end;
+    	end; 
 
-end;
+	end;
 
 
-  *next step7- 1b.7 - discordant abortion clean up;
- If outassgn_pt1  in ('SAB' 'IAB'  ) then do;
-  *if both (SAB+IAB +/- UAB);
-  if sab ne '000' and iab ne '000' then outassgn_pt2='SAB'; *3b.7.1;
-   
-   *if just one;
-   else do ;*1b.7.2 - go back to table 1 for abortion;
-         if sab NE '000' /*and uab Not In ('000')*/ then outassgn_pt2='SAB'; 
-    else if iab NE '000' /*and uab Not In ('000') */ then outassgn_pt2='IAB';
-/*    else if sab='000' and iab='000' and uab not in ('000') then Outcome_assigned4_clean='UAB';*/
-  end;
- end;
+  	*next step7- 1b.7 - discordant abortion clean up;
+ 	If outassgn_pt1  in ('SAB' 'IAB' 'UAB' ) then do; 
+	*CDL: ADDED UDL to be consistent with alg 3. Might be some edge cases captured. - Chaned documentation;
 
- *1b8 - final outcome assignment;
- If outassgn_pt2 ne '' then Outcome_Assigned4 = outassgn_pt2;
- else outcome_assigned4 = outassgn_pt1;
+		*CDL: MODIFIED to be more straightforward;
+		if SAB ne ('000') and IAB ne ('000') then outassgn_pt2='SAB'; *Deal w discordant abortion outcomes first;
+				else if SAB ne ('000') then outassign_pt2='SAB';
+				else if IAB ne ('000') then outassign_pt2 = 'IAB';
+		*NOTE: Some may have meds for IAB or SAB because they had concordant codes on an encounter whihc were 
+				rolled up to the group level;
 
- *use the highest codetype for outcome type assigned to group;
-    *only UDL has procedures, all other delivery types are diagnosis code based;
-    *AEM is only dx based but all other outcomes have multiple possible code types;
-    *for abortion if both SAB and IAB then otucome would be UAB so ok to use Abortion value (covers all);
-      IF Outcome_Assigned4 in ('LBM' 'LBS' 'UDL' 'SB' 'MLS') then Outcome_Assigned_Codetype4= Delivery;
+/*  		*if both (SAB+IAB +/- UAB);*/
+/*  		if sab ne '000' and iab ne '000' then outassgn_pt2='SAB'; *3b.7.1;*/
+/*   */
+/*   		*if just one;*/
+/*   		else do ;*1b.7.2 - go back to table 1 for abortion;*/
+/*        	if sab NE '000' then outassgn_pt2='SAB'; */
+/*    		else if iab NE '000' then outassgn_pt2='IAB';*/
+/*    		else if sab='000' and iab='000' and uab not in ('000') then Outcome_assigned4_clean='UAB';*/
+/*  		end;*/
+ 	end;

```

Comment:
Included a more straightforward version of this code than what is commented out.